regolith/helpers/a_todohelper.py

```diff
@@ -0,0 +1,157 @@
+from regolith.fsclient import _id_key
+from regolith.tools import (
+    all_docs_from_collection,
+    get_pi_id,
+    document_by_value,
+)
+
+TARGET_COLL = "people"
+ALLOWED_IMPORTANCE = [0, 1, 2]
+
+
+def subparser(subpi):
+    subpi.add_argument("description", help="the description of the to_do task. If the description has more than one "
+                                           "word, please enclose it in quotation marks.",
+                       default=None)
+    subpi.add_argument("due_date",
+                       help="Due date of the task. Either enter a date in format YYYY-MM-DD or an "
+                            "integer. Integer 5 means 5 days from the "
+                            "begin_date. "
+                       )
+    subpi.add_argument("-i", "--id", help="ID of the to-do list's owner. The default id is saved in user.json. ")
+    subpi.add_argument("-b", "--begin_date",
+                       help="Begin date of the task in format YYYY-MM-DD. Default is today."
+                       )
+    subpi.add_argument("-d", "--duration",
+                       help="The estimated duration of the task in minutes. Default is 60 ",
+                       default=60
+                       )
+    subpi.add_argument("-p", "--importance",
+                       help=f"The importance of the task from {ALLOWED_IMPORTANCE}. Default is 1.",
+                       default=1
+                       )
+    subpi.add_argument("-n", "--notes", nargs="+", help="Additional notes for this task. Each note should be enclosed "
+                                                        "in quotation marks.")
+
+    return subpi
+
+
+class TodoAdderHelper(DbHelperBase):
+    """Helper for adding a to_do task to people.yml
+    """
+    # btype must be the same as helper target in helper.py
+    btype = "a_todo"
+    needed_dbs = [f'{TARGET_COLL}']
+
+    def construct_global_ctx(self):
+        """Constructs the global context"""
+        super().construct_global_ctx()
+        gtx = self.gtx
+        rc = self.rc
+        if "groups" in self.needed_dbs:
+            rc.pi_id = get_pi_id(rc)
+
+        rc.coll = f"{TARGET_COLL}"
+        rc.database = rc.databases[0]["name"]
+        gtx[rc.coll] = sorted(
+            all_docs_from_collection(rc.client, rc.coll), key=_id_key
+        )
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+        gtx["zip"] = zip
+
+    def db_updater(self):
+        rc = self.rc
+        if not rc.id:
+            try:
+                rc.id = rc.default_user_id
+            except AttributeError:
+                print(
+                    "Please set default_user_id in '~/.config/regolith/user.json',\
+                     or you need to enter your group id in the command line")
+                return
+        try:
+            person = document_by_value(all_docs_from_collection(rc.client, "people"), "_id", rc.id)["name"]
```

Comment:
I think this will be very slow.  Could you try and benchmark this against ``get_one`` or sthg like that.  This is loading a whole huge collection and iterating through it item by item to match an id instead of just trying to retrieve an id and either succeeding or failing.